homeassistant/components/influxdb/config_flow.py

```diff
@@ -0,0 +1,157 @@
+    CONF_INCLUDE,
+    CONF_PASSWORD,
+    CONF_PATH,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_TOKEN,
+    CONF_URL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import AbortFlow
+
+from . import DOMAIN, get_influx_connection
+from .const import (
+    CONF_API_VERSION,
+    CONF_BUCKET,
+    CONF_COMPONENT_CONFIG,
+    CONF_COMPONENT_CONFIG_DOMAIN,
+    CONF_COMPONENT_CONFIG_GLOB,
+    CONF_DB_NAME,
+    CONF_DEFAULT_MEASUREMENT,
+    CONF_IGNORE_ATTRIBUTES,
+    CONF_MEASUREMENT_ATTR,
+    CONF_ORG,
+    CONF_OVERRIDE_MEASUREMENT,
+    CONF_PRECISION,
+    CONF_RETRY_COUNT,
+    CONF_SSL_CA_CERT,
+    CONF_TAGS,
+    CONF_TAGS_ATTRIBUTES,
+    DEFAULT_API_VERSION,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _validate_influxdb_connection(
+    hass: HomeAssistant, data: dict[str, Any]
+) -> dict[str, str] | None:
+    """Validate connection to influxdb."""
+
+    try:
+        influx = await hass.async_add_executor_job(get_influx_connection, data, True)
+    except ConnectionError as ex:
+        _LOGGER.error(ex)
+        return {"base": "cannot_connect"}
+    except Exception:
+        _LOGGER.exception("Unknown error")
+        return {"base": "unknown"}
+
+    influx.close()
+
+    return None
+
+
+class InfluxDBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for InfluxDB."""
+
+    VERSION = 1
+
+    async def _abort_if_host_database_configured(
+        self,
+        data: dict[str, Any],
+        options: dict[str, Any],
+    ) -> None:
+        """Test if host and database are already configured."""
+        host = data.get(CONF_HOST)
+        database = data.get(CONF_DB_NAME)
+        bucket = data.get(CONF_BUCKET)
+
+        for entry in self._async_current_entries():
+            if (
+                entry.data[CONF_API_VERSION] == DEFAULT_API_VERSION
+                and entry.data[CONF_HOST] == host
+                and entry.data.get(CONF_DB_NAME) == database
+            ) or (
+                entry.data[CONF_API_VERSION] != DEFAULT_API_VERSION
+                and entry.data[CONF_HOST] == host
+                and entry.data.get(CONF_BUCKET) == bucket
+            ):
+                if data is not None and options is not None:
+                    changed = self.hass.config_entries.async_update_entry(
+                        entry,
+                        data={**entry.data, **data},
+                        options={**entry.options, **options},
+                    )
+                    if changed and entry.state in (
+                        config_entries.ConfigEntryState.LOADED,
+                        config_entries.ConfigEntryState.SETUP_RETRY,
+                    ):
+                        self.hass.async_create_task(
+                            self.hass.config_entries.async_reload(entry.entry_id)
+                        )
+                raise AbortFlow("already_configured")
```

Comment:
this doesn't sound like aborting, rather like updating and aborting. Ideally we don't do that anymore and have users use reconfigure for this (unless I am missing something why we do this)

Reply:
The problem is that with the amount of options, we need quite a complex option flow for reconfigure. With this behavior, it always updates from the yaml.
If we don’t want this intermediate behavior, either this PR increases in size (which I wouldn’t do) or we don’t push to dev but an intermediate branch to push all the changes incrementally to have a full config flow and push that in the end to dev.

Reply:
I think like we spoke before, there are also options that are nearly impossible to properly get into a config flow, so I think a good first step that would be mergable would be to just setup the connection via the UI and do the rest via YAML (+ maybe some simple options).

I kinda want to keep the import and user step in the same PR, mostly because the way you do one, affects the other (schema, required data, etc). So if we would merge this as is, and we have to change stuff, we have a harder time in having a clear overview in why we did what we did.

And nevertheless, I am committed on getting this through, so don't be scared to be hit with a "this is too big" :)

Reply:
Ok, let’s do the import and user step together.

Reply:
Oh and to continue on reconfiguring, I think that is something that we can do quite fast after implementing this one, so the options are still changeable. Everything not required for the connection (the optional things) can be set using the options flow. I think it would be nice to be in here, but I also don't have the full scope of influxdb in my mind, so if that is indeed one beast, I think adding reconfigure and options for a followup with intermediary branch is nice